connector/connector.py

```diff
@@ -316,6 +326,29 @@ def get_connector_unit(self, base_class):
         return self.backend.get_class(base_class, self.session,
                                       self.model_name)(self)
 
+    @classmethod
+    def create_environment(cls, backend_record, session, model,
+                           connector_env=None):
```

Comment:
Yeah, I was thinking about the same. But "explicit is better than implicit", so as you said, the API would be more complicated.